mock/mock_test.go

```diff
@@ -1159,6 +1161,15 @@ func Test_Arguments_Bool(t *testing.T) {
 
 }
 
+func Test_ArgumentMatcher_Matches_Nil(t *testing.T) {
+
+	var matcher = argumentMatcher{fn: reflect.ValueOf(func(v interface{}) bool {
+		return v == nil
+	})}
+	assert.True(t, matcher.Matches(nil))
+
+}
```

Comment:
I would like to:
- Test the public API rather than the internals (create a mock with a `MatchedBy` expectation)
- Test each of the types that had issues: interfaces, maps, slices, func, channels, and pointers
- Test that it still panics when receiving `nil` when any other type is expected